projects/libyaml/libyaml_emitter_fuzzer.c

```diff
@@ -0,0 +1,296 @@
+                    event_from->data.document_start.version_directive,
+                    event_from->data.document_start.tag_directives.start,
+                    event_from->data.document_start.tag_directives.end,
+                    event_from->data.document_start.implicit);
+
+        case YAML_DOCUMENT_END_EVENT:
+            return yaml_document_end_event_initialize(event_to,
+                    event_from->data.document_end.implicit);
+
+        case YAML_ALIAS_EVENT:
+            return yaml_alias_event_initialize(event_to,
+                    event_from->data.alias.anchor);
+
+        case YAML_SCALAR_EVENT:
+            return yaml_scalar_event_initialize(event_to,
+                    event_from->data.scalar.anchor,
+                    event_from->data.scalar.tag,
+                    event_from->data.scalar.value,
+                    event_from->data.scalar.length,
+                    event_from->data.scalar.plain_implicit,
+                    event_from->data.scalar.quoted_implicit,
+                    event_from->data.scalar.style);
+
+        case YAML_SEQUENCE_START_EVENT:
+            return yaml_sequence_start_event_initialize(event_to,
+                    event_from->data.sequence_start.anchor,
+                    event_from->data.sequence_start.tag,
+                    event_from->data.sequence_start.implicit,
+                    event_from->data.sequence_start.style);
+
+        case YAML_SEQUENCE_END_EVENT:
+            return yaml_sequence_end_event_initialize(event_to);
+
+        case YAML_MAPPING_START_EVENT:
+            return yaml_mapping_start_event_initialize(event_to,
+                    event_from->data.mapping_start.anchor,
+                    event_from->data.mapping_start.tag,
+                    event_from->data.mapping_start.implicit,
+                    event_from->data.mapping_start.style);
+
+        case YAML_MAPPING_END_EVENT:
+            return yaml_mapping_end_event_initialize(event_to);
+    }
+
+    return 0;
+}
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  if (size < 2) return 0;
+
+  yaml_parser_t parser;
+  yaml_emitter_t emitter;
+  yaml_event_t event;
+  unsigned char buffer[BUFFER_SIZE+1];
+  size_t written = 0;
+  yaml_event_t events[MAX_EVENTS];
+  size_t event_number = 0;
+  bool done = false;
+  int count = 0;
+  int error = 0;
+  bool is_canonical = data[0] & 1;
+  bool is_unicode = data[1] & 1;
+  memset(buffer, 0, BUFFER_SIZE+1);
+  memset(events, 0, MAX_EVENTS*sizeof(yaml_event_t));
+
+  if(!yaml_parser_initialize(&parser))
+    return 0;
+
+  yaml_parser_set_input_string(&parser, data, size);
+  if(!yaml_emitter_initialize(&emitter)) {
+    yaml_parser_delete(&parser);
+    return 0;
+  }
+
+  if (is_canonical) {
+      yaml_emitter_set_canonical(&emitter, 1);
+  }
+  if (is_unicode) {
+      yaml_emitter_set_unicode(&emitter, 1);
+  }
+  yaml_emitter_set_output_string(&emitter, buffer, BUFFER_SIZE, &written);
```

Comment:
maybe use yaml_emitter_set_output to not have an output buffer size that is too small, you can extend a buffer in the function that you give it.